lib/Conversion/TorchToTosa/TorchBackendLegalToTosaCustom.cpp

```diff
@@ -0,0 +1,201 @@
+    if (auto unrealizedCastOp = dyn_cast_or_null<UnrealizedConversionCastOp>(
+            inputOperand.getDefiningOp())) {
+      torchInputOperand = unrealizedCastOp.getInputs()[0];
+    }
+    // Handle the special case where input operand is an argument to the module
+    // function
+    if (!torchInputOperand.getDefiningOp())
+      return inputOperand;
+
+    return TypeSwitch<Operation *, Value>(torchInputOperand.getDefiningOp())
+        .Case<Torch::ConstantBoolOp>([&](Operation *boolOperand) -> Value {
+          bool boolConstAttr;
+          if (matchPattern(boolOperand, m_TorchConstantBool(&boolConstAttr))) {
+            return tosa::getConstTensor<int64_t>(rewriter, backendLegalOp,
+                                                 boolConstAttr, {})
+                .value();
+          }
+          return nullptr;
+        })
+        .Case<Torch::ConstantIntOp>([&](Operation *intOperand) -> Value {
+          int64_t intConstAttr;
+          if (matchPattern(intOperand, m_TorchConstantInt(&intConstAttr))) {
+            return tosa::getConstTensor<int64_t>(rewriter, backendLegalOp,
+                                                 intConstAttr, {})
+                .value();
+          }
+          return nullptr;
+        })
+        .Case<Torch::ConstantFloatOp>([&](Operation *floatOperand) -> Value {
+          double floatConstAttr;
+          if (matchPattern(floatOperand,
+                           m_TorchConstantFloat(&floatConstAttr))) {
+            return tosa::getConstTensor<float>(rewriter, backendLegalOp,
+                                               floatConstAttr, {})
+                .value();
+          }
+          return nullptr;
+        })
+        .Case<Torch::PrimListConstructOp>(
+            [&](Operation *intListConstructOperand) -> Value {
+              SmallVector<int64_t> intConstListAttr;
+              if (matchPattern(intListConstructOperand,
+                               m_TorchConstantIntList(intConstListAttr))) {
+                return tosa::getConstTensor<int64_t>(
+                           rewriter, backendLegalOp, intConstListAttr,
+                           {static_cast<int64_t>(intConstListAttr.size())})
+                    .value();
+              }
+              return nullptr;
+            })
+        .Case<Torch::ConstantNoneOp>([&](Operation *noneOperand) -> Value {
+          auto noneCustomOp = rewriter.create<tosa::CustomOp>(
+              backendLegalOp->getLoc(),
+              RankedTensorType::get({}, rewriter.getIntegerType(1)),
+              rewriter.getStringAttr("torch.none"), ValueRange{});
+          return noneCustomOp.getResult(0);
+        })
+        .Default([&](Operation *defaultOperand) { return inputOperand; });
+  }
+
+  LogicalResult
+  matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const final {
+
+    for (auto customOpName : customOps) {
```

Comment:
This pattern gets applied one op at a time, so there should not be a need for iterating through all of `customOps`. Instead, `customOps` should be stored as a set, and this pattern should begin with:

```c++
if (!customOps.contains(op->getName().getStringRef())
  return rewriter.notifyMatchFailure(...);
```

Reply:
Done, updated `customOps` to be a `SetVector` instead of an `ArrayRef` to do look-ups and avoid looping on the custom operations instead.